Fixed Issues related to 3rd Party Applications which extend Native Types

AndiDittrich opened the pull request:
Related to my [Post on WordPress.org](https://wordpress.org/support/topic/getting-an-error-because-jquery-isnt-loading?replies=7#post-8083921)

Here is my fix that your plugin works with 3rd Party Applications which extend Native Types.
The `for ..in` loop to generate the asset section is replaced by [jQuery.each](http://api.jquery.com/jquery.each/) - this avoids conflicts with object properties (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in)

Additionally the `getAssetURL` function checks the asset before processing

NateWr commented on assets/js/aqm.js:
```diff
@@ -67,13 +67,22 @@ jQuery(document).ready(function ($) {
-					}
-				}
-			}
+ 			var T = this;
```
Could you rename `T` to `self`? I just prefer that over `T`.

NateWr commented:
Thanks Andi. The `adm.js` file is actually compiled by the grunt script. Would you be willing to apply your changes to `/assets/src/js/aqm-panel.js`?

If you've got Grunt set up already, you can run `grunt watch` before you make the changes to auto compile. Or you can run the `jshint`, `contact` and `uglify` commands manually after-the-fact.

Or if you don't have Grunt set up, just modify `aqm-panel.js` and I'll compile everything after I merge.

Thanks!

AndiDittrich commented:
HI Nate,
i've applied the requested changes as well as fixed the initial check for `aqm` and `aqmData` object.
a new grunt task-alias `dist` is also added to run the grunt tasks within a single step, without watching them.

NateWr commented:
Looks great @AndiDittrich! And thanks for adding the easy grunt task. 

Would you be able to convert your changes to tabs instead of spaces, so we can keep it consistent across the project? Then I'm happy to merge it.

AndiDittrich commented:
no problem, i have changed it right now.
additionally, `getAssetURL` now returns boolean type **false** on error instead on string. i've changed the related code:

```js
var url = this.getAssetURL( asset );
if ( url !== false ) {
   html += '<a href="' + url + '" target="_blank" class="view">' + aqm.strings.view + '</a>';
}
```

i hope that this change is ok ?

NateWr commented:
Looks great @AndiDittrich, thanks! I'll prepare a release now.
